### PR TITLE
Adding definition for rgw dns name - Closes #437

### LINKED ISSUE #437: [doc] 19.3.1.1  Additional Notes
### Type of report
   - [ x ] bug
   - [ ] feature request

### Source Document
19.3.1.1  Additional Notes:

https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#sec-ceph-rgw-configuration-notes

### Summary

Explain the reasoning behind `rgw dns name`. Its primary use is to allow clients using vhost style buckets to be used. 
vhost style access refers to `<bucketname>.<s3-endpoint>/<object-path>` in comparison to a `<s3-endpoint>/<bucket>/<object>`
This also means that the site must have the dns configured correctly to correctly reroute dns requests of the <bucketname>.<s3-endpoint> style to RGW correctly. This `s3-endpoint` must correspond to the `rgw dns name` configurable

### PR DESCRIPTION
**Pull request by asettle**

### CODE REVIEW

File: xml/runtime_config.xml [hunk 1]
```diff
@@ -1038,12 +1038,15 @@ rgw_frontends = beast port=7480
...
       <term>rgw_dns_name</term>
       <listitem>
-       <para>
-        If the parameter <literal>rgw dns name</literal> is added to the
-        <filename>ceph.conf</filename>, make sure that the S3 client is
-        configured to direct requests at the endpoint specified by <literal>rgw
-        dns name</literal>.
-       </para>
+        <para>
+          Allows clients to use vhost style buckets to be used.
```
Comment (lproven) [inline xml/runtime_config.xml:1042]:
```suggestion
          Allows clients to use <literal>vhost</literal>-style buckets.
```

File: xml/runtime_config.xml [hunk 2]
```diff
@@ -1038,12 +1038,15 @@ rgw_frontends = beast port=7480
...
+        </para>
+        <para>
+          If the parameter <literal>rgw dns name</literal> is added to the
+          <filename>ceph.conf</filename>, make sure that the S3 client is
+          configured to direct requests at the endpoint specified by <literal>rgw
```
Comment (lproven) [inline xml/runtime_config.xml:1047]:
```suggestion
          configured to direct requests to the endpoint specified by <literal>rgw
```

Comment (lproven) [reply xml/runtime_config.xml:1047]: (This is just a suggestion -- if "at" is the more normal usage, then leave it.

**Comment by theanalyst** [comment]
Can we define what `vhost style buckets` mean  somewhere, the point being people don't really know what it means for the most case and that has been the source of many bugs

**Comment by asettle** [comment]
> Can we define what `vhost style buckets` mean somewhere, the point being people don't really know what it means for the most case and that has been the source of many bugs

Are you refering to this definition, `vhost style access refers to <bucketname>.<s3-endpoint>/<object-path> in comparison to a <s3-endpoint>/<bucket>/<object>`?

**Comment by theanalyst** [comment]
> > Can we define what `vhost style buckets` mean somewhere, the point being people don't really know what it means for the most case and that has been the source of many bugs
> 
> Are you refering to this definition, `vhost style access refers to <bucketname>.<s3-endpoint>/<object-path> in comparison to a <s3-endpoint>/<bucket>/<object>`?

Yes, is that already defined like that somewhere?

File: xml/runtime_config.xml [hunk 3]
```diff
@@ -1038,12 +1038,20 @@ rgw_frontends = beast port=7480
...
-        configured to direct requests at the endpoint specified by <literal>rgw
-        dns name</literal>.
-       </para>
+        <para>
+          Allows clients to use <literal>vhost</literal>-style buckets.
+        </para>
+        <para>
+          A <literal>vhost</literal> style access refers to
+          <replaceable>bucketname</replaceable>.<replaceable>s3-endpoint</replaceable>/<replaceable>object-path</replaceable>.
+          This is in comparison to the path style access: <replaceable>s3-endpoint</replaceable>/<replaceable>bucket</replaceable>/<replaceable>object</replaceable>
```
Comment (lproven) [inline xml/runtime_config.xml:1047]:
```suggestion
          <literal>vhost</literal>-style access refers to the use of 
          <replaceable>bucketname</replaceable>.<replaceable>s3-endpoint</replaceable>/<replaceable>object-path</replaceable>.
          This is in comparison to  <literal>path</literal>-style access: <replaceable>s3-endpoint</replaceable>/<replaceable>bucket</replaceable>/<replaceable>object</replaceable>
```

File: xml/runtime_config.xml [hunk 4]
```diff
@@ -1038,12 +1038,20 @@ rgw_frontends = beast port=7480
...
+        </para>
+        <para>
+          If the parameter <literal>rgw dns name</literal> is added to the
```
Comment (lproven) [inline xml/runtime_config.xml:1050]:
```suggestion
          If the parameter <literal>rgw dns name</literal> is added to
```

**Pull request merged**